### PR TITLE
O2 breaks enhancements and cleanup, planner options UI

### PR DESCRIPTION
**Pull request by sfuchs79**
### Describe the pull request:

- [x] Bug fix
- [x] Functional change
- [ ] New feature
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

After the last fixes around the o2 breaks in the planner I found two other issue:
1. Now the minimum gas switch time for the switch to oxy is also suppressed if O2 breaks are switched on but no O2 breaks happen because O2 deco time is too short (<12min) for O2 breaks. Finally we decided to really always suppress minimum switch time for change to O2 but to also reflect this in the UI.
Plus does some cleanup of (unused) variables.

2. Disabling "last stop at 6m/20ft" still breaks the whole O2 breaks timing calculation with the current implementation. My proposal to fix this is to disable the whole O2 breaks feature if "last stop at 6m" is not activated. IMHO this makes sense - please correct me if I'm wrong! Benefit also would be reducing the number of possible combinations of planner options in this specific area (testing effort!).

All this comes also with two slightly off topic commit to even more consistently enable/disable some planner option and UI elements based on selected deco mode and change the position of the dive type selection in the planner UI. IMHO these ones are no brainers.

### CODE REVIEW

**Comment by sfuchs79** [comment]
Ok, I'm optimistic today and do announce that this is ready from my side ;-)
But it really needs some review please...

**Comment by sfuchs79** [comment]
Too optimistic...
A minute ago I found another small oddity - almost negligible but I will try to fix it tomorrow.

File: core/planner.c [hunk 1]
```diff
@@ -1017,32 +1015,36 @@ bool plan(struct diveplan *diveplan, struct dive *dive, int timestep, struct dec
...
-								o2breaking  = true;
-								o2time = 0;
-								if (is_final_plan)
-									plan_add_segment(diveplan, laststoptime, depth, current_cylinder, po2, false);
-								previous_point_time = clock + laststoptime;
-								current_cylinder = breakfrom_cylinder;
-								gas = dive->cylinder[current_cylinder].gasmix;
-								breaktime = -1;
-							}
+					} else if (o2break_next) {
```
Comment (atdotde) [inline core/planner.c:1027]: Coding style: Please write this as
} else {
    if (...

Comment (glance-) [reply core/planner.c:1027]: I don't agree here.

I think you rather should lift up the inner if statement into this instead.

I also think there is too much inline code here, so if its possible, we should look at lifting those out to separate helpers to keep the general code flow easier on the eyes.

**Comment by sfuchs79** [comment]
@atdotde @glance- 
Sure I will take care about any coding style items but I have to postpone this because this thing here is still driving me crazy from pure functional point of view.

I'm talking about only one commit from the above three, the "O2 breaks code enhancements and cleanup" thing.
We always have the issue with doing the minimum gas switch time or not. With Roberts proposal we always suppressed the minimum gas switch time as soon as option "O2 breaks" is active, no matter if O2 breaks happen or not. Then I made a proposal where I was able to fix this for one case but for the case when we have no deco at all the result was still wrong.
Now I pushed another proposal but this still doesn't work correctly. Now we always consider the minimum gas switch time once also when we switch to oxy and O2 breaks are on. If we then don't do O2 breaks results is ok. But if we do O2 breaks we again have the issue that we would need to somehow "undo" the minimum gas switch time leg or compensate it somehow because otherwise the last leg during O2 breaking is too short by one time the minimum gas switch time.
Any ideas?

Btw there would be a easy solution but it would mean a changed behavior and would need documentation even in the UI: Simply never do consider the minimum gas switch time for the switch to oxy/above 7m. What do you think about this?

**Comment by atdotde** [comment]
I was aware of this problem when I wrote the commit that made Subsurface ignore the minimal gas switching time for the 6m O2 stop. My rationale was that this was an academic situation anyway: It only bites of you are doing a dive using O2 as deco gas but where the 6m stop is so short that it is shorter than the minimal gas switching time. I don't think anybody would do such a dive in real life (if the deco is so short, you wouldn't bother carrying an O2 stage).

Of course, it would be possible to introduce an additional variable to note that the minimal gas switching time has already been added so that the O2 time is then not 12min but reduced by that time (of course only for the first o2 leg) but decided against it since I did not consider this important enough (and admittedly was in a rush). But, yes, my way to fix this would be to reduce the 12*60 sec by the minimal time if that had already been added (figure out a way to determine that with the least effort).

**Comment by sfuchs79** [comment]
> I was aware of this problem when I wrote the commit that made Subsurface ignore the minimal gas switching time for the 6m O2 stop. My rationale was that this was an academic situation anyway: It only bites of you are doing a dive using O2 as deco gas but where the 6m stop is so short that it is shorter than the minimal gas switching time.

Yes, fully agree. Then let's keep it simple and go back to the solution (your solution) where minimum gas switch time is NEVER considered for switches to oxy on 6m! I would just reflect this in UI and maybe also documentation. I would change the label in the UI to something like "Minimum gas switch time for >6m/20ft". I can do this during the next days...
Ok for you?

File: core/planner.c (hunk 1 above)
Comment (dirkhh) [reply core/planner.c:1027]: @atdotde - no, we do allow `} else if (...) {`

**Comment by sfuchs79** [comment]
Ok, let's be optimistic again:
This should work now as expected and discussed.

Now we also can discuss about coding style items but please don't be too strict with me this time because I have currently so much open things to finish and so little time ;-(

**Pull request merged**